src/library_fs.js

```diff
@@ -27,13 +27,14 @@ mergeInto(LibraryManager.library, {
     ],
   $FS__postset: function() {
     // TODO: do we need noFSInit?
-    addAtInit('if (!Module["noFSInit"] && !FS.init.initialized) FS.init();');
-    addAtMain('FS.ignorePermissions = false;');
+    addAtInit('FS.ignorePermissions = false; if (!Module["noFSInit"] && !FS.init.initialized) FS.init();');
```

Comment:
I think these two may need to be reordered - the FS creates read-only directories etc. during init, IIRC.

Reply:
Wouldn't many tests fail if that was needed?

Reply:
Yes... so I guess I was wrong.